app/src/main/java/eu/kanade/tachiyomi/ui/setting/database/ClearDatabaseController.kt

```diff
@@ -0,0 +1,163 @@
+package eu.kanade.tachiyomi.ui.setting.database
+
+import android.annotation.SuppressLint
+import android.app.Dialog
+import android.os.Bundle
+import android.view.*
+import androidx.core.view.*
+import androidx.core.view.WindowInsetsCompat.Type.systemBars
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.google.android.material.dialog.MaterialAlertDialogBuilder
+import com.google.android.material.floatingactionbutton.ExtendedFloatingActionButton
+import eu.davidea.flexibleadapter.FlexibleAdapter
+import eu.davidea.flexibleadapter.Payload
+import eu.kanade.tachiyomi.R
+import eu.kanade.tachiyomi.databinding.ClearDatabaseControllerBinding
+import eu.kanade.tachiyomi.ui.base.controller.DialogController
+import eu.kanade.tachiyomi.ui.base.controller.FabController
+import eu.kanade.tachiyomi.ui.base.controller.NucleusController
+import eu.kanade.tachiyomi.util.system.toast
+import eu.kanade.tachiyomi.util.view.liftAppbarWith
+import eu.kanade.tachiyomi.util.view.scrollViewWith
+
+class ClearDatabaseController :
+    NucleusController<ClearDatabaseControllerBinding, ClearDatabasePresenter>(),
+    FlexibleAdapter.OnItemClickListener,
+    FlexibleAdapter.OnUpdateListener,
+    FabController {
+
+    private var recycler: RecyclerView? = null
```

Comment:
also not needed to define

Reply:
did you mean both `FlexibleAdapter.OnUpdateListener` and `private var recycler: RecyclerView? = null`? removing 
`private var recycler: RecyclerView? = null` causes https://github.com/Jays2Kings/tachiyomiJ2K/blob/c916512b67caf22e96080389ab8462b4214becf6/app/src/main/java/eu/kanade/tachiyomi/ui/setting/database/ClearDatabaseController.kt#L63 to be unresolved

Reply:
just `private var recycler: RecyclerView? = null`